### ISSUE TITLE
ICE when incrementing a complex variable

### ISSUE THREAD

**Issue by rainers** [ldc-developers/ldc#1806]
Compiling with current master under Windows:

```
void foo()
{
	cdouble c = 1.0 + 1.0i;
	c++;
}
```

yields:
```
Both operands to a binary operator are not of the same type!
  %3 = fadd { double, double } %2, ppc_fp128 0xM3FF00000000000000000000000000000
LLVM ERROR: Broken function found, compilation aborted!
```
Same for cfloat. `c = c + 1` works.

**Comment by kinke** [comment]
`ppc_fp128` :D

I can only assume that all other platforms fail too (and for all precisions) as LLVM's `fadd` obviously doesn't work with complex numbers.

**Comment by LemonBoy** [comment]
Nice, the structure holding the complex number (which is just a `double[2]`) is recognized as a `ppc_fp128` here [1].
It might be better to gate this behind a check of the target triple.

[1] https://github.com/ldc-developers/ldc/blob/master/gen/tollvm.cpp#L439

**Comment by kinke** [comment]
Fixed by #1809.
